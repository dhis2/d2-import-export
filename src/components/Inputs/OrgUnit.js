import React from 'react'
import i18n from '@dhis2/d2-i18n'

import { Field } from '../Field/Field'
import { Label } from '../Field/Label'
import { OrgUnitTree } from '../FinalFormComponents/OrgUnitTree'

export const ORG_UNIT_KEY = 'orgUnit'
export const ORG_UNIT_DEFAULT_VALUE = []

export const OrgUnit = () => (
    <Field>
        <Label>{i18n.t('Organisation unit')}</Label>
        <OrgUnitTree name={ORG_UNIT_KEY} label={i18n.t('Organisation unit')} />
    </Field>
)
