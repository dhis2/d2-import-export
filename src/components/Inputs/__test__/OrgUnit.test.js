import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { OrgUnit } from '../OrgUnit'

jest.mock('react-final-form')

describe('Input component - OrgUnit', () => {
    it('should render correctly', () => {
        const file = shallow(<OrgUnit />)

        expect(toJson(file)).toMatchSnapshot()
    })
})
