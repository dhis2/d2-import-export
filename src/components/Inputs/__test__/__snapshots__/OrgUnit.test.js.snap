// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component - OrgUnit should render correctly 1`] = `
<Field>
  <Label>
    Organisation unit
  </Label>
  <OrgUnitTree
    label="Organisation unit"
    multiple={true}
    name="orgUnit"
    selected={Array []}
  />
</Field>
`;
